src/engine/source/api/src/test/handlers.cpp

```diff
@@ -458,6 +467,113 @@ api::Handler sessionsDelete(shared_ptr<Router> router, shared_ptr<Catalog> catal
+        const auto errorMsg = !eRequest.has_name()    ? std::make_optional("Missing /session name")
+                        : !eRequest.has_event() ? std::make_optional("Missing /event")
+                                                : std::nullopt;
+
+        if (errorMsg.has_value())
+        {
+            return ::api::adapter::genericError<ResponseType>(errorMsg.value());
+        }
+
+        // Set optional parameters
+        debugMode = eRequest.has_debugmode() ? eRequest.debugmode() : eTest::DebugMode::ONLY_OUTPUT;
+
+        protocolQueue =  eRequest.has_protocolqueue() ? eRequest.protocolqueue() : TEST_DEFAULT_PROTOCOL_QUEUE;
+
+        protocolLocation = eRequest.has_protocollocation() ? eRequest.protocollocation() : TEST_DEFAULT_PROTOCOL_LOCATION;
+
+        // Set debug mode
+        auto debugModes = [](eTest::DebugMode debugModeValue) -> router::DebugMode
+        {
+            std::unordered_map<eTest::DebugMode, router::DebugMode> debugModeMap = {
+                {eTest::DebugMode::ONLY_OUTPUT, router::DebugMode::ONLY_OUTPUT},
+                {eTest::DebugMode::OUTPUT_AND_TRACES, router::DebugMode::OUTPUT_AND_TRACES},
+                {eTest::DebugMode::OUTPUT_AND_TRACES_WITH_DETAILS, router::DebugMode::OUTPUT_AND_TRACES_WITH_DETAILS}};
+            if (debugModeMap.find(debugModeValue) != debugModeMap.end())
+            {
+                return debugModeMap[debugModeValue];
+            }
+            else
+            {
+                return router::DebugMode::ONLY_OUTPUT;
+            }
+        };
+
+        // Get session
+        auto& sessionManager = SessionManager::getInstance();
+        auto session = sessionManager.getSession(eRequest.name());
+        if (!session.has_value())
+        {
+            return ::api::adapter::genericError<ResponseType>(
+                fmt::format("Session '{}' was not found", eRequest.name()));
+        }
+
+        // Event in Wazuh format
+        const auto eventFormat = fmt::format(TEST_EVENT_CONTENT_FORMAT,
+                                       protocolQueue,
+                                       protocolLocation,
+                                       eRequest.event().string_value(),
+                                       eRequest.name());
+        auto ev = std::make_shared<json::Json>(eventFormat.c_str());
+
+        const auto enqueueEventError = router->enqueueEvent(std::move(ev));
+        if (enqueueEventError.has_value())
+        {
+            return ::api::adapter::genericError<ResponseType>(enqueueEventError.value().message);
+        }
+
+        ResponseType eResponse;
+
+        // Get payload (output and traces)
+        auto payload = router->getData(session.value().getPolicyName(), debugModes(debugMode));
+
+        if (std::holds_alternative<base::Error>(payload))
+        {
+            return ::api::adapter::genericError<ResponseType>(std::get<base::Error>(payload).message);
+        }
```

Comment:
```suggestion
        // Get payload (output and traces)
        auto payload = router->getData(session.value().getPolicyName(), debugModes(debugMode));

        if (std::holds_alternative<base::Error>(payload))
        {
            return ::api::adapter::genericError<ResponseType>(std::get<base::Error>(payload).message);
        }

        ResponseType eResponse;
```
Basically moving `ResponseType eResponse;` to the bottom, closer to its utilization.